src/main/java/org/home/implementation/SimpleFileMessageStorageImpl.java

```diff
@@ -0,0 +1,117 @@
+package org.home.implementation;
+
+import org.home.interfaces.MessageStorageInterface;
+import org.home.model.FileMessage;
+import org.home.model.Message;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+
+//@Component("secondary")
+public class SimpleFileMessageStorageImpl implements MessageStorageInterface {
+    String QFileName;
+    BufferedWriter oo;
+    BufferedReader io;
+    public SimpleFileMessageStorageImpl(String qname) {
+        QFileName = "Queue-"+qname+".txt";
+        try {
+            oo = new BufferedWriter(new FileWriter(QFileName));
+            //oo.write("");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        finally{
+            if(oo!=null)
+            {
+                try {
+                    oo.close();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
+
+    @Override
+    public void storeMessage(Message message) {
+        try {
+            oo = new BufferedWriter(new FileWriter(QFileName,true));
+            String msg = message.getKey()+ "," + message.getBody();
+            oo.write(msg);
+            oo.write("\n");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        finally{
+            if(oo!=null)
+            {
+                try {
+                    oo.close();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
+
+    @Override
+    public Message fetchMessage() {
+        boolean message_read = false;
+        BufferedWriter oos = null;
+        Message m=null;
+        try {
+            io = new BufferedReader(new FileReader(QFileName));
+            String fileLine = null;
```

Comment:
No need to open every-time. 
Learn to use caching.
Save this buffered-reader in an in-memory map and re-use it every-time you call it.